ENIGMAsystem/SHELL/Platforms/SDL/Event.cpp

```diff
@@ -0,0 +1,219 @@
+      case SDL_AUDIODEVICEREMOVED:
+        //audioDeviceRemoved(&e);
+        break;
+      case SDL_QUIT:
+        quit(&e);
+        return 1;
+      case SDL_FINGERMOTION:
+        //fingerMotion(&e);
+        break;
+      case SDL_FINGERDOWN:
+        //fingerDown(&e);
+        break;
+      case SDL_FINGERUP:
+        //fingerUp(&e);
+        break;
+      case SDL_MULTIGESTURE:
+        //multigesture(&e);
+        break;
+      case SDL_DOLLARGESTURE:
+        //dollarGesture(&e);
+        break;
+      case SDL_DOLLARRECORD:
+        //dollarRecord(&e);
+        break;
+      case SDL_DROPFILE:
+        //dropFile(&e);
+        break;
+      case SDL_DROPTEXT:
+        //dropText(&e);
+        break;
+      case SDL_DROPBEGIN:
+        //dropBegin(&e);
+        break;
+      case SDL_DROPCOMPLETE:
+        //dropEnd(&e);
+        break;
+    }
+  }
+
+  return 0;
+}
+
+void SDL_Event_Handler::windowEvent(const SDL_Event *event) {
+  switch (event->window.event) {
+    case SDL_WINDOWEVENT_SHOWN:
+      //windowShown(event);
+      break;
+    case SDL_WINDOWEVENT_HIDDEN:
+      //windowHidden(event);
+      break;
+    case SDL_WINDOWEVENT_EXPOSED:
+      //windowExposed(event);
+      break;
+    case SDL_WINDOWEVENT_MOVED:
+      //windowMoved(event);
+      break;
+    case SDL_WINDOWEVENT_RESIZED:
+      //windowResized(event);
+      break;
+    case SDL_WINDOWEVENT_SIZE_CHANGED:
+      break;  // will trigger above event too
+    case SDL_WINDOWEVENT_MINIMIZED:
+      //windowMinimized(event);
+      break;
+    case SDL_WINDOWEVENT_MAXIMIZED:
+      //windowMaximized(event);
+      break;
+    case SDL_WINDOWEVENT_RESTORED:
+      //windowRestored(event);
+      break;
+    case SDL_WINDOWEVENT_ENTER:
+      //windowEntered(event);
+      break;
+    case SDL_WINDOWEVENT_LEAVE:
+      //windowLeave(event);
+      break;
+    case SDL_WINDOWEVENT_FOCUS_GAINED:
+      //windowFocusGain(event);
+      break;
+    case SDL_WINDOWEVENT_FOCUS_LOST:
+      //windowFocusLost(event);
+      break;
+    case SDL_WINDOWEVENT_CLOSE:
+      //windowClose(event);
+      break;
+    case SDL_WINDOWEVENT_TAKE_FOCUS:
+      //windowTakeFocus(event);
+      break;
+    case SDL_WINDOWEVENT_HIT_TEST:
+      //windowHitTest(event);
+      break;
+    default:
+      //unkownEvent(event);
+      break;
+  }
+}
+
+void SDL_Event_Handler::keyboardDown(const SDL_Event *event) {
+  int key = event->key.keysym.sym;
+  enigma::last_keybdstatus[key] = enigma::keybdstatus[key];
+  enigma::keybdstatus[key] = true;
+
+  //if (key == SDLK_BACKSPACE && !enigma_user::keyboard_string.empty()) enigma_user::keyboard_string.pop_back();
+}
+
+void SDL_Event_Handler::keyboardUp(const SDL_Event *event) {
+  int key = event->key.keysym.sym;
+  enigma::last_keybdstatus[key] = enigma::keybdstatus[key];
+  enigma::keybdstatus[key] = false;
+}
+
+void SDL_Event_Handler::textInput(const SDL_Event *event) {
+  /*enigma_user::keyboard_string += event->text.text;
```

Comment:
Did this not work properly?

Reply:
I'm not sure I think I had compiler error before some of my other changes. It might be another variable I need to migrate that I missed. I forgotten I had commented this